### ISSUE TITLE
Changing Text Language setting for a file should trigger ability to Save

### ISSUE THREAD

**Issue by Frozsoul2019** [sirjuddington/SLADE#999]
**SLADE Version**: (3.1.4)
**Operating System**: Windows

**Issue Details**:
When you change only the Text Language setting in an open file, and not any of the file's content, SLADE doesn't provide the option to save the file (and therefore save the new Text Language setting).  However if you modify any of the content in the file as well (e.g. add and remove a space character), SLADE will allow you to save all changes made to the file, which will also include any changes to the Text Language specified.  The solution is that anytime the Text Language setting is changed, it should trigger the option to Save the file (and therefore the Text Language setting)

**Comment by Unstabler2018** [comment]
It actually appears that the Text Language setting does not save when the file is closed.  Requesting that the Text Language setting be able to be saved.

**Comment by Gaerzi** [comment]
This creates the problem of where and how to save this metadata.

**Comment by Unstabler2018** [comment]
It appears that this metadata is being saved somehow?  For example, the DECORATE files within Brutal Doom's PK3 files all have TXT file extension but are recognized in SLADE as Type=Decorate and Text Language=ZDoom Decorate.  However very similar files with TXT file extension within my custom PK3 file are only recognized in SLADE as Type=Text and Text Language=None.

**Comment by Gaerzi** [comment]
No, there are other mechanisms at play.  SLADE will correctly identify entries that are included by another entry, notably.

**Comment by Unstabler2018** [comment]
Ah, thank you for the clarification.  I was working on a patch file for BD and I did not have a DECORATE entry that #included the various TXT files, because they are already #included in the original mod.  By including the DECORATE entry in my patch file the TXT files are now correctly identified by SLADE.  The only difficulty is that by loading both the mod and the patch, the DECORATE files are now #included twice, which results in a ton of GZDoom script warnings.

**Comment by Gaerzi** [comment]
Yes, the situation of an entry being included from a different archive is not taken into account.

In your case, which is a bit unusual, I'd suggest just renaming the root DECORATE entry something meaningless, like MECORATE (I like using M for modified when I do this kind of things) when saving to test, to avoid the double-loading.

**Comment by Unstabler2018** [comment]
Thank you, that is probably the best solution my case.  The original issue is still outstanding though for active SLADE sessions, as changes to a file's Text Language setting is saved until you re-open the file.
